app/controllers/case_logs_controller.rb

```diff
@@ -20,22 +20,19 @@ def edit
     render :edit
   end
 
-  def next_page
+  def submit_form
     form = Form.new(2021, 2022)
-    @case_log = CaseLog.find(params[:case_log_id])
-    previous_page = params[:previous_page]
+    @case_log = CaseLog.find(params[:id])
+    previous_page = params[:case_log][:previous_page]
     questions_for_page = form.questions_for_page(previous_page).keys
     answers_for_page = page_params(questions_for_page).select { |k, _v| questions_for_page.include?(k) }
-    @case_log.update!(answers_for_page)
-    next_page = form.next_page(previous_page)
-    redirect_path = if next_page == :check_answers
-                      subsection = form.subsection_for_page(previous_page)
-                      "case_log_#{subsection}_check_answers_path"
-                    else
-                      "case_log_#{next_page}_path"
-                    end
-
-    redirect_to(send(redirect_path, @case_log))
+    if @case_log.update(answers_for_page)
```

Comment:
This will mean it will validate for tenant_age even when submitting for tenant_code - this is something I can work out on my branch however thought it worth mentioning

Reply:
Yeah this isn't meant to be a complete implementation by any means, just the basis for your work to start really.